src/ui/Base/Component.ts

```diff
@@ -454,7 +454,14 @@ export class Component extends BaseComponent {
   static resolveRoot(element: HTMLElement): HTMLElement {
     Assert.exists(element);
     const resolvedSearchInterface = Component.resolveBinding(element, SearchInterface);
-    return resolvedSearchInterface ? resolvedSearchInterface.element : document.body;
+    if (resolvedSearchInterface) {
```

Comment:
Would `$$(element).closest('.CoveoSearchInterface')` do the trick for this whole function ?

Reply:
doesn't work for IE https://developer.mozilla.org/en-US/docs/Web/API/Element/closest would need a polyfill

Reply:
That's a custom function that we created in `Dom.ts`, not the native browser function.

Reply:
for `resolveBinding`, I believe that it would, yes. The only exception is that `resolveBinding` also checks the descendants, but I would be very surprised if there's a case where that's necessary.

Reply:
edit: this is already what `resolveBinding` calls, but with a small chance of being breaking. I don't see much value in changing it, so I'll leave it like this.